polaris/io.py

```diff
@@ -164,6 +164,13 @@ def symlink(target, link_name, overwrite=True):
         Whether to replace an existing link if one already exists
     """
 
+    # make the directory that the link is in if it doesn't exist
+    directory = os.path.dirname(os.path.abspath(link_name))
+    try:
+        os.makedirs(directory)
+    except OSError:
+        pass
```

Comment:
Perhaps this would be better?
```suggestion
    directory = os.path.dirname(os.path.abspath(link_name))
    try:
        os.makedirs(directory)
    except FileExistsError:
        pass
```
Something like this is the preferred way try to make a directory, but to do nothing if it already exists.  It avoids a race condition that can happen if you first check if the directory exists and then create it.